service/controller/resource/s3bucket/create.go

```diff
@@ -156,7 +156,9 @@ func (r *Resource) newCreateChange(ctx context.Context, obj, currentState, desir
 	var createState []BucketState
 	for _, bucket := range desiredBuckets {
 		if !containsBucketState(bucket.Name, currentBuckets) {
-			createState = append(createState, bucket)
+			// in case any of the bucket is missing
+			// rerun all code for all buckets to update bucket logging as well
+			createState = desiredBuckets
```

Comment:
this is maybe too magic,  basically, when this function found that some bucket is missing it will rerun `CreateBucket` for all buckets, this will ensure that when we create a new  logging bucket, the access logs of the  ignition bucket is set to he new one

it's only executed once during the update

but like to know some opinions @giantswarm/team-firecracker-engineers

Reply:
I think this makes sense. I would just put a `break` statement after assigning the desired state to the current state. Then we safe some CPU cycles.

Reply:
ack